### ISSUE TITLE
Target allocator tests are failing intermittently

### ISSUE THREAD

**Issue by jpkrohling** [open-telemetry/opentelemetry-operator#396]
The CI seems to be broken at the moment and seems to have started with the target allocator commit. 

cc @Aneurysm9, @Raul9595, @dashpole 

```
=== CONT  kuttl/harness/smoke-targetallocator
    logger.go:42: 07:53:00 | smoke-targetallocator/2-install | test step failed 2-install
    case.go:361: failed in step 2-install
    case.go:363: --- StatefulSet:kuttl-test-stirred-pigeon/stateful-collector
        +++ StatefulSet:kuttl-test-stirred-pigeon/stateful-collector
        @@ -1,9 +1,200 @@
         apiVersion: apps/v1
         kind: StatefulSet
         metadata:
        +  annotations:
        +    opentelemetry-operator-config/sha256: 7b3bc1a44a6c240b33070c9681ffd023240a422c628ca76e2924df402da12492
        +    prometheus.io/path: /metrics
        +    prometheus.io/port: "8888"
        +    prometheus.io/scrape: "true"
        +  labels:
        +    app.kubernetes.io/component: opentelemetry-collector
        +    app.kubernetes.io/instance: kuttl-test-stirred-pigeon.stateful
        +    app.kubernetes.io/managed-by: opentelemetry-operator
        +    app.kubernetes.io/name: stateful-collector
        +    app.kubernetes.io/part-of: opentelemetry
        +  managedFields:
        +  - apiVersion: apps/v1
        +    fieldsType: FieldsV1
        +    fieldsV1:
        +      f:status:
        +        f:collisionCount: {}
        +        f:currentReplicas: {}
        +        f:currentRevision: {}
        +        f:observedGeneration: {}
        +        f:replicas: {}
        +        f:updateRevision: {}
        +        f:updatedReplicas: {}
        +    manager: kube-controller-manager
        +    operation: Update
        +    subresource: status
        +    time: "2021-08-19T07:52:30Z"
        +  - apiVersion: apps/v1
        +    fieldsType: FieldsV1
        +    fieldsV1:
        +      f:metadata:
        +        f:annotations:
        +          .: {}
        +          f:opentelemetry-operator-config/sha256: {}
        +          f:prometheus.io/path: {}
        +          f:prometheus.io/port: {}
        +          f:prometheus.io/scrape: {}
        +        f:labels:
        +          .: {}
        +          f:app.kubernetes.io/component: {}
        +          f:app.kubernetes.io/instance: {}
        +          f:app.kubernetes.io/managed-by: {}
        +          f:app.kubernetes.io/name: {}
        +          f:app.kubernetes.io/part-of: {}
        +        f:ownerReferences:
        +          .: {}
        +          k:{"uid":"c1b52214-5263-4774-9f3a-954d76eb72cd"}: {}
        +      f:spec:
        +        f:podManagementPolicy: {}
        +        f:replicas: {}
        +        f:revisionHistoryLimit: {}
        +        f:selector: {}
        +        f:serviceName: {}
        +        f:template:
        +          f:metadata:
        +            f:labels:
        +              .: {}
        +              f:app.kubernetes.io/component: {}
        +              f:app.kubernetes.io/instance: {}
        +              f:app.kubernetes.io/managed-by: {}
        +              f:app.kubernetes.io/name: {}
        +              f:app.kubernetes.io/part-of: {}
        +          f:spec:
        +            f:containers:
        +              k:{"name":"otc-container"}:
        +                .: {}
        +                f:args: {}
        +                f:image: {}
        +                f:imagePullPolicy: {}
        +                f:name: {}
        +                f:resources: {}
        +                f:terminationMessagePath: {}
        +                f:terminationMessagePolicy: {}
        +                f:volumeMounts:
        +                  .: {}
        +                  k:{"mountPath":"/conf"}:
        +                    .: {}
        +                    f:mountPath: {}
        +                    f:name: {}
        +                  k:{"mountPath":"/usr/share/default-volume"}:
        +                    .: {}
        +                    f:mountPath: {}
        +                    f:name: {}
        +            f:dnsPolicy: {}
        +            f:restartPolicy: {}
        +            f:schedulerName: {}
        +            f:securityContext: {}
        +            f:serviceAccount: {}
        +            f:serviceAccountName: {}
        +            f:terminationGracePeriodSeconds: {}
        +            f:volumes:
        +              .: {}
        +              k:{"name":"otc-internal"}:
        +                .: {}
        +                f:configMap:
        +                  .: {}
        +                  f:defaultMode: {}
        +                  f:items: {}
        +                  f:name: {}
        +                f:name: {}
        +        f:updateStrategy:
        +          f:rollingUpdate:
        +            .: {}
        +            f:partition: {}
        +          f:type: {}
        +        f:volumeClaimTemplates: {}
        +    manager: manager
        +    operation: Update
        +    time: "2021-08-19T07:52:30Z"
           name: stateful-collector
           namespace: kuttl-test-stirred-pigeon
        +  ownerReferences:
        +  - apiVersion: opentelemetry.io/v1alpha1
        +    blockOwnerDeletion: true
        +    controller: true
        +    kind: OpenTelemetryCollector
        +    name: stateful
        +    uid: c1b52214-5263-4774-9f3a-954d76eb72cd
        +spec:
        +  podManagementPolicy: Parallel
        +  replicas: 1
        +  revisionHistoryLimit: 10
        +  selector:
        +    matchLabels:
        +      app.kubernetes.io/component: opentelemetry-collector
        +      app.kubernetes.io/instance: kuttl-test-stirred-pigeon.stateful
        +      app.kubernetes.io/managed-by: opentelemetry-operator
        +      app.kubernetes.io/name: stateful-collector
        +      app.kubernetes.io/part-of: opentelemetry
        +  serviceName: stateful-collector
        +  template:
        +    metadata:
        +      creationTimestamp: null
        +      labels:
        +        app.kubernetes.io/component: opentelemetry-collector
        +        app.kubernetes.io/instance: kuttl-test-stirred-pigeon.stateful
        +        app.kubernetes.io/managed-by: opentelemetry-operator
        +        app.kubernetes.io/name: stateful-collector
        +        app.kubernetes.io/part-of: opentelemetry
        +    spec:
        +      containers:
        +      - args:
        +        - --config=/conf/collector.yaml
        +        image: otel/opentelemetry-collector:0.31.0
        +        imagePullPolicy: IfNotPresent
        +        name: otc-container
        +        resources: {}
        +        terminationMessagePath: /dev/termination-log
        +        terminationMessagePolicy: File
        +        volumeMounts:
        +        - mountPath: /conf
        +          name: otc-internal
        +        - mountPath: /usr/share/default-volume
        +          name: default-volume
        +      dnsPolicy: ClusterFirst
        +      restartPolicy: Always
        +      schedulerName: default-scheduler
        +      securityContext: {}
        +      serviceAccount: stateful-collector
        +      serviceAccountName: stateful-collector
        +      terminationGracePeriodSeconds: 30
        +      volumes:
        +      - configMap:
        +          defaultMode: 420
        +          items:
        +          - key: collector.yaml
        +            path: collector.yaml
        +          name: stateful-collector
        +        name: otc-internal
        +  updateStrategy:
        +    rollingUpdate:
        +      partition: 0
        +    type: RollingUpdate
        +  volumeClaimTemplates:
        +  - apiVersion: v1
        +    kind: PersistentVolumeClaim
        +    metadata:
        +      creationTimestamp: null
        +      name: default-volume
        +    spec:
        +      accessModes:
        +      - ReadWriteOnce
        +      resources:
        +        requests:
        +          storage: 50Mi
        +      volumeMode: Filesystem
        +    status:
        +      phase: Pending
         status:
        -  readyReplicas: 1
        +  collisionCount: 0
        +  currentReplicas: 1
        +  currentRevision: stateful-collector-566dcf84b6
        +  observedGeneration: 1
           replicas: 1
        +  updateRevision: stateful-collector-566dcf84b6
        +  updatedReplicas: 1
         
        
    case.go:363: resource StatefulSet:kuttl-test-stirred-pigeon/stateful-collector: .status.readyReplicas: key is missing from map
    logger.go:42: 07:53:00 | smoke-targetallocator | smoke-targetallocator events from ns kuttl-test-stirred-pigeon:
    logger.go:42: 07:53:00 | smoke-targetallocator | 2021-08-19 07:52:30 +0000 UTC	Normal	PersistentVolumeClaim default-volume-stateful-collector-0		WaitForFirstConsumer	waiting for first consumer to be created before binding		
    logger.go:42: 07:53:00 | smoke-targetallocator | 2021-08-19 07:52:30 +0000 UTC	Normal	PersistentVolumeClaim default-volume-stateful-collector-0		ExternalProvisioning	waiting for a volume to be created, either by external provisioner "rancher.io/local-path" or manually created by system administrator		
    logger.go:42: 07:53:00 | smoke-targetallocator | 2021-08-19 07:52:30 +0000 UTC	Normal	StatefulSet.apps stateful-collector		SuccessfulCreate	create Claim default-volume-stateful-collector-0 Pod stateful-collector-0 in StatefulSet stateful-collector success		
    logger.go:42: 07:53:00 | smoke-targetallocator | 2021-08-19 07:52:30 +0000 UTC	Normal	StatefulSet.apps stateful-collector		SuccessfulCreate	create Pod stateful-collector-0 in StatefulSet stateful-collector successful		
    logger.go:42: 07:53:00 | smoke-targetallocator | 2021-08-19 07:52:30 +0000 UTC	Normal	Pod stateful-targetallocator-5c9fdcc85c-fsgkc		Scheduled	Successfully assigned kuttl-test-stirred-pigeon/stateful-targetallocator-5c9fdcc85c-fsgkc to kind-control-plane		
    logger.go:42: 07:53:00 | smoke-targetallocator | 2021-08-19 07:52:30 +0000 UTC	Normal	ReplicaSet.apps stateful-targetallocator-5c9fdcc85c		SuccessfulCreate	Created pod: stateful-targetallocator-5c9fdcc85c-fsgkc		
    logger.go:42: 07:53:00 | smoke-targetallocator | 2021-08-19 07:52:30 +0000 UTC	Normal	Deployment.apps stateful-targetallocator		ScalingReplicaSet	Scaled up replica set stateful-targetallocator-5c9fdcc85c to 1		
    logger.go:42: 07:53:00 | smoke-targetallocator | 2021-08-19 07:52:32 +0000 UTC	Normal	Pod stateful-targetallocator-5c9fdcc85c-fsgkc.spec.containers{ta-container}		Pulling	Pulling image "quay.io/opentelemetry/target-allocator:0.1.0"		
    logger.go:42: 07:53:00 | smoke-targetallocator | 2021-08-19 07:52:37 +0000 UTC	Normal	PersistentVolumeClaim default-volume-stateful-collector-0		Provisioning	External provisioner is provisioning volume for claim "kuttl-test-stirred-pigeon/default-volume-stateful-collector-0"		
    logger.go:42: 07:53:00 | smoke-targetallocator | 2021-08-19 07:52:38 +0000 UTC	Normal	Pod stateful-targetallocator-5c9fdcc85c-fsgkc.spec.containers{ta-container}		Pulled	Successfully pulled image "quay.io/opentelemetry/target-allocator:0.1.0" in 6.354160959s		
    logger.go:42: 07:53:00 | smoke-targetallocator | 2021-08-19 07:52:38 +0000 UTC	Normal	Pod stateful-targetallocator-5c9fdcc85c-fsgkc.spec.containers{ta-container}		Created	Created container ta-container		
    logger.go:42: 07:53:00 | smoke-targetallocator | 2021-08-19 07:52:38 +0000 UTC	Normal	Pod stateful-targetallocator-5c9fdcc85c-fsgkc.spec.containers{ta-container}		Started	Started container ta-container		
    logger.go:42: 07:53:00 | smoke-targetallocator | 2021-08-19 07:52:53 +0000 UTC	Normal	PersistentVolumeClaim default-volume-stateful-collector-0		ProvisioningSucceeded	Successfully provisioned volume pvc-a650fc51-bed0-479b-897e-9d022276f13d		
    logger.go:42: 07:53:00 | smoke-targetallocator | 2021-08-19 07:52:54 +0000 UTC	Normal	Pod stateful-collector-0		Scheduled	Successfully assigned kuttl-test-stirred-pigeon/stateful-collector-0 to kind-control-plane		
    logger.go:42: 07:53:00 | smoke-targetallocator | 2021-08-19 07:52:56 +0000 UTC	Normal	Pod stateful-collector-0.spec.containers{otc-container}		Pulled	Container image "otel/opentelemetry-collector:0.31.0" already present on machine		
    logger.go:42: 07:53:00 | smoke-targetallocator | 2021-08-19 07:52:56 +0000 UTC	Normal	Pod stateful-collector-0.spec.containers{otc-container}		Created	Created container otc-container		
    logger.go:42: 07:53:00 | smoke-targetallocator | 2021-08-19 07:52:56 +0000 UTC	Normal	Pod stateful-collector-0.spec.containers{otc-container}		Started	Started container otc-container		
    logger.go:42: 07:53:00 | smoke-targetallocator | Deleting namespace: kuttl-test-stirred-pigeon
=== CONT  kuttl
    harness.go:399: run tests finished
    harness.go:508: cleaning up
    harness.go:563: removing temp folder: ""
--- FAIL: kuttl (84.95s)
    --- FAIL: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/smoke-simplest (10.32s)
        --- PASS: kuttl/harness/daemonset-features (14.44s)
        --- PASS: kuttl/harness/smoke-sidecar (18.46s)
        --- PASS: kuttl/harness/smoke-statefulset (25.45s)
        --- PASS: kuttl/harness/targetallocator-features (27.83s)
        --- PASS: kuttl/harness/statefulset-features (28.47s)
        --- FAIL: kuttl/harness/smoke-targetallocator (36.89s)
FAIL
```

**Comment by Raul9595** [comment]
The tests have been flaky since the target allocator was added. From what I noticed in the logs, its mostly the case of the volumeClaimTemplates for the StatefulSet being stuck in `Pending` state. Assuming this to the case, we can add an additional TestAssert command to ensure the StatefulSet is up and running before we proceed with the assertion of the other parameters. Something like the one below may be able to solve this issue.

`kubectl rollout status --watch --timeout=300s statefulset/name-of-statefulset`
